internal/cloudapi/v2/v2.go

```diff
@@ -397,22 +360,110 @@ func (h *apiHandlers) PostCompose(ctx echo.Context) error {
 			t.ImageName = fmt.Sprintf("composer-api-%s", uuid.New().String())
 		}
 
-		imageRequest.target = t
+		irTarget = t
 	default:
 		return HTTPError(ErrorUnsupportedImageType)
 	}
 
-	id, err := h.server.workers.EnqueueOSBuild(imageRequest.arch, &worker.OSBuildJob{
-		Manifest: imageRequest.manifest,
-		Targets:  []*target.Target{imageRequest.target},
-		Exports:  imageRequest.exports,
-	})
+	manifestJobID, err := h.server.workers.EnqueueManifestJobByID(&worker.ManifestJobByID{}, depsolveJobID)
+	if err != nil {
+		return HTTPErrorWithInternal(ErrorEnqueueingJob, err)
+	}
+
+	id, err := h.server.workers.EnqueueOSBuildAsDependency(arch.Name(), &worker.OSBuildJob{
+		Manifest: nil, // distro.Manifest([]byte{}),
+		Targets:  []*target.Target{irTarget},
+		Exports:  imageType.Exports(),
+	}, manifestJobID)
 	if err != nil {
 		return HTTPErrorWithInternal(ErrorEnqueueingJob, err)
 	}
 
 	ctx.Logger().Infof("Job ID %s enqueued for operationID %s", id, ctx.Get("operationID"))
 
+	// start 1 go routing which requests datajob type
+	go func(workers *worker.Server, manifestJobID uuid.UUID, b *blueprint.Customizations, options distro.ImageOptions, repos []rpmmd.RepoConfig, seed int64) {
+		// wait until job is in a pending state
+		var token uuid.UUID
+		var dynArgs []json.RawMessage
+		var err error
+		for {
+			_, token, _, _, dynArgs, err = workers.RequestJobById(context.Background(), "", manifestJobID)
+			if err == jobqueue.ErrNotPending {
+				logrus.Infof("manifest job %v not pending, waiting for depsolve job to finish", manifestJobID)
+				time.Sleep(time.Second)
+				continue
+			}
+			if err != nil {
+				logrus.Errorf("Error requesting manifest job: %v", err)
+				return
+			}
+			break
+		}
+
+		finishJob := func(jr worker.ManifestJobByIDResult) {
```

Comment:
I think I would rather use `defer`. See how it's done elsewhere: https://github.com/osbuild/osbuild-composer/blob/main/cmd/osbuild-worker/jobimpl-osbuild.go#L64